### DISCUSSION TITLE
Mongo Model with objects ?

### DISCUSSION THREAD

**Discussion by gbkwiatt** [strapi/strapi#6341] [Legacy Topics]
I am trying to understand  how strapi works. I want to use mongoDb, but my collection model require to have nested objects. I guess I could use 'json' type for model, but then nested object won't be validated ? Is that even possible with strapi ? I am just not understanding it ? 

```json
  "attributes": {
    "test": {
      "type": "json",
      "required": true,
      "nestedObject": {
        "value1": {
          "type": "string"
        },
        "value2": {
          "type": "string"
        }
      }
    }
  }
    },
}
```

**Comment by gbkwiatt** [comment]
I did try to have a look on components, which kind of makes it work like that, but it creates separate collection for each of those objects and making relation by id ? Is it not making non-sql an sql database  ?

**Comment by lauriejim** [comment]
Hello! I'm not sure to completely understand what you are trying to do.
Can you please precise with a real use case example.
Thank you.

**Comment by gbkwiatt** [comment]
There is no way to create nested object in the collection model. 
for example I want to store something like this in my mongo collection: 
```json
{
 "someProperty": "value",
 "nestedObject": {
       "nestedProperty": "value
 }
}
```

to do it in strapi, I have to create component/collection, that will have `nestedProperty`  property and then reference ID from my other document, but that means in my database I have extra collection to store my nested documents.

Mongoose allows models made with nested objects e.g.: 
```js
{
  name: {
    type: String,
    required: [true, 'Please provide category Name'],
  },
  description: String,
  nestedObject: {
    type: {
        nestedValue1: String,
        nestedValue2: Number
    }
  }
}
```

**Comment by lauriejim** [comment]
OKay, no that not possible to do that in Strapi by default, 
after that I suggest to add a custom code in the life cycle function that will check this in the before create and update functions.
https://strapi.io/documentation/v3.x/concepts/models.html#lifecycle-hooks
